libs/common-types/src/impls.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+use super::*;
+use frame_support::traits::{EnsureOrigin, EnsureOriginWithArg};
+use frame_system::RawOrigin;
+use sp_std::marker::PhantomData;
+
+type AccountId = u64;
+
+/// The OrmlAssetRegistry::AuthorityOrigin impl
+pub struct AuthorityOrigin<
```

Comment:
Can we name this somehow that it describes its purpose?

Reply:
To my understanding, we can't. Because we overwrite the standard implementation. I will add additional code comments to make the purpose more clear!

Reply:
But aren't we implementing on our own struct? As long as we satisfy the bounds, I guess we can name it as we like.

Reply:
Yeah, we can. It's just that I want to be sure we stay as close to the real code as possible. We use the exact same naming, so if we want to reason about the logic, we don't run into a completely different name and have to wonder: Why did we use this in the first place?

I am open for a change, but for me, this doesn't make the logic nor the reasoning about it any simpler.